### PR TITLE
[test stats] record invoking file in test cases

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80793
* __->__ #80792

See the comment for explanation.

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
should change the comment to reflect the truth, but lgtm

File: tools/stats/upload_test_stats.py [hunk 1]
```diff
@@ -38,6 +38,18 @@ def parse_xml_report(
...
         case["job_id"] = job_id
+
+        # [invoking file]
+        # The name of the file that the test is located in is not necessarily
+        # the same as the name of the file that invoked the test.
+        # For example, `test_jit.py` calls into multiple other test files (e.g.
+        # jit/test_dce.py). For sharding/test selection purposes, we want to
+        # record the file that invoked the test.
+        #
+        # To do this, we leverage an implementation detail of xmlrunner, which
```
[the remark this answers is not in the document]
Comment (suo) [reply tools/stats/upload_test_stats.py:49]: ahhhh lol

[the remark this answers is not in the document]
Comment (suo) [reply tools/stats/upload_test_stats.py:49]: I'll change the comment

**Pull request closed without merging**